tests/test_session.py

```diff
@@ -323,3 +325,20 @@ def test_read_case_using_lightweight_mode():
 
 def test_help_does_not_throw(new_solver_session):
     help(new_solver_session.file.read)
+
+
+@pytest.mark.skipif(
+    os.getenv("PYFLUENT_LAUNCH_CONTAINER") == "1", reason="tests Fluent launch scenario"
+)
+def test_recover_grpc_error_from_launch_error(monkeypatch: pytest.MonkeyPatch):
+    orig_parse_server_info_file = session._parse_server_info_file
+
+    def mock_parse_server_info_file(file_name):
+        ip, port, password = orig_parse_server_info_file(file_name)
+        return ip, port - 1, password  # provide wrong port
+
+    monkeypatch.setattr(session, "_parse_server_info_file", mock_parse_server_info_file)
+    with pytest.raises(LaunchFluentError) as ex:
+        solver = pyfluent.launch_fluent()
+    # grpc.RpcError -> RuntimeError -> LaunchFluentError
+    assert ex.value.__context__.__context__.code() == grpc.StatusCode.UNAVAILABLE
```

Comment:
Had an idea for the test 

Advantages of suggestion below:
- Much quicker, almost instant, as it does not actually launch Fluent
- Does not leave zombies behind
- Should achieve roughly the same objective, as `connect_to_fluent` uses mostly the same methods as `launch_fluent` in the background, and the error status code raised is the same 
- Tested to work in Windows and Linux with/without containers

Disadvantages: 
- Not testing the `launch_fluent` method itself
```suggestion
def test_recover_grpc_error_from_connection_error():
    with pytest.raises(RuntimeError) as ex:
        pyfluent.connect_to_fluent(ip="127.0.0.1", port=50000, password="abcdefg")
    assert ex.value.__context__.code() == grpc.StatusCode.UNAVAILABLE
```